src/main/scala/com/codacy/analysis/cli/command/analyse/AnalyseExecutor.scala

```diff
@@ -102,24 +103,31 @@ class AnalyseExecutor(toolInput: Option[String],
         logger.error(e)(s"Failed analysis for ${tool.name}")
     }
 
-    uploader.fold[Future[Either[String, Unit]]]({ message =>
+    uploader.fold[Future[M[Unit]]]({ message =>
       logger.warn(message)
-      Future.successful(().asRight[String])
+      Future.successful(monadError.pure(()))
     }, { upload =>
       for {
         results <- Future.fromTry(result)
         upl <- upload.sendResults(tool.name, results)
-      } yield upl
+      } yield eitherToM(upl)
     })
   }
 
-  private def sequenceWithFixedLeft[A](left: A)(eitherIterable: Seq[Either[A, Unit]]): Either[A, Unit] = {
-    eitherIterable
-      .foldLeft[Either[A, Unit]](Right(())) { (acc, either) =>
-        acc.flatMap(_ => either)
-      }
-      .left
-      .map(_ => left)
+  private def eitherToM[A](either: Either[String, A]): M[A] = {
```

Comment:
Can't we create an M in the first place instead of converting an Either to M?

Reply:
Yep. Just did a single file so it wasn't a massive PR as a demonstration of this syntax.